### PR TITLE
Fix and Update to gateway v4

### PR DESCRIPTION
**Pull request by PixeLInc**
Changed to use the IP received in the Ready payload.
Added and changed to use the heartbeat interval from the Hello payload.
Removed heartbeat interval from Ready payload.
Switch to gateway v4

### CODE REVIEW

File: client.go [hunk 1]
```diff
@@ -162,7 +164,8 @@ func (c *Client) runForever() {
 		}
 
 		var err error
-		c.ws, _, err = websocket.DefaultDialer.Dial("wss://"+strings.TrimSuffix(info.Endpoint, ":80"), nil)
+		host := "wss://" + strings.TrimSuffix(info.Endpoint, ":80") + ":443/?v=4"
```
Comment (b1naryth1ef) [inline client.go]: Is the port even really necessary here?

File: client.go [hunk 2]
```diff
@@ -253,13 +266,13 @@ func (c *Client) runHeartbeater(closer chan struct{}, interval time.Duration) {
 	}
 }
 
-func (c *Client) runUDP(closer chan struct{}, port int, ssrc uint32) {
+func (c *Client) runUDP(closer chan struct{}, connection_ip string, port int, ssrc uint32) {
```
Comment (b1naryth1ef) [inline client.go]: `connectionIP`

**Review by b1naryth1ef — APPROVED** [review]
LGTM, I'll need to find time after Christmas to test :)

File: client.go (hunk 1 above)
Comment (PixeLInc) [reply client.go]: It doesn't seem to be, removed it.

**Comment by b1naryth1ef** [comment]
Thanks!

**Pull request merged**